src/ast/passes/semantic_analyser.cpp

```diff
@@ -346,6 +346,14 @@ void SemanticAnalyser::visit(Builtin &builtin)
         LOG(ERROR, builtin.loc, err_)
             << "The func builtin can not be used with '"
             << attach_point->provider << "' probes";
+
+      if ((type == ProbeType::kretprobe || type == ProbeType::uretprobe) &&
+          !bpftrace_.feature_->has_helper_get_func_ip()) {
+        LOG(ERROR, builtin.loc, err_)
+            << "The 'func' builtin is not available for " << type
+            << "s on kernels without the get_func_ip BPF feature. Consider "
+               "using 'probe' instead.";
```

Comment:
nit: can we say either 'kprobe' or 'uprobe' depending on the context ('probe' by itself isn't a thing) ?

Reply:
I think it means the `probe` builtin ;-)

Reply:
Hopefully it's clearer now!